app/src/main/java/org/secuso/privacyfriendlyfoodtracker/database/ApplicationDatabase.java

```diff
@@ -72,14 +73,46 @@ public void onCreate(@NonNull SupportSQLiteDatabase db) {
                                         Log.e("ApplicationDatabase", e.getMessage());
                                     }
                                 }
-                            }).build();
+                            })
+                            .addMigrations(MIGRATION_1_2).addMigrations(MIGRATION_2_3).build();
 
                 }
             }
         }
         return sInstance;
     }
 
+    /*
+    Migrate the database so it contains columns with carbs,fat,satfat etc.
+     */
+    static final Migration MIGRATION_1_2 = new Migration(1, 2) {
+        @Override
+        public void migrate(SupportSQLiteDatabase database) {
+            String[] newFields = new String[]{"carbs","sugar", "protein", "fat","satFat"};
+            for(String field : newFields) {
+                database.execSQL("ALTER TABLE Product "
+                        + " ADD COLUMN "+ field +" REAL NOT NULL default 0");
+            }
+        }
+    };
+
+    /***
+     * Migrate database so it also contains micro nutriments.
+     */
+    static final Migration MIGRATION_2_3 = new Migration(2, 3) {
```

Comment:
I think you are right about the database version. I did not look into this for a long time though.
I can look into the npe in like one or two weeks. Do you have more information on the exception?

Reply:
Hi, I also took a look and fixed some bugs: 
- At some places product.key was used instead of offKey which leads to Zero values in database products
- The click listener in search products was weird implemented. If there was a reason for it I don't get it. I cleaned it up a little which fixed the crash mentioned above.
How can i contribute my changes? This will be my first contribution to an Open Source Project on GitHub.
I have done the changes in the forked repositpory of @conrad57 an have them only local on my machine.
@Kamuno  And there are merge conflicts. Which branch do I have to merge into this to resolve the conflicts?